### PR TITLE
Adds example of Dask-ML's HyperbandSearchCV

### PR DESCRIPTION
**Pull request by stsievert**
**What does this PR implement?**
It implements an example of Dask-ML's `HyperbandSearchCV`. It mentions

* how to determine inputs for `HyperbandSearchCV`
* model requirements
* actual use of Hyperband

**Reference issues/PRs**
In the future, I'll add more information on

* initial exploration (https://github.com/dask/dask-ml/pull/532)
* the verbose flat (https://github.com/dask/dask-ml/pull/528)
* the blog post I have drafted (https://stsievert.com/drafts/hyperband)
* the Dask model selection documentation (https://github.com/dask/dask-ml/pull/432)

### CODE REVIEW

File: machine-learning/hyperparam-opt.ipynb [hunk 1]
```diff
@@ -0,0 +1,510 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from distributed import Client\n",
+    "client = Client(\"localhost:8786\")\n",
```
Comment (mrocklin) [inline machine-learning/hyperparam-opt.ipynb]: This notebook should be runnable on its own without external setup.  We usually create a Client with a specific number of workers and threads-per-worker fully specified with a setup that we know works well.  It's also worth noting that these are likely to run on Binder pods that are pretty low-power.  Typically we pretend that we have something like 4 cores and a couple gigabytes of memory.  

You might want to look at some of the other examples to see how we tend to manage this.

File: machine-learning/hyperparam-opt.ipynb [hunk 2]
```diff
@@ -0,0 +1,510 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**This example will walk through**\n",
+    "\n",
+    "* how to determine the input parameters\n",
+    "* how to use the a particular algorithm implemented in `HyperbandSearchCV`"
```
Comment (TomAugspurger) [inline machine-learning/hyperparam-opt.ipynb]: I don't follow this bullet.

File: machine-learning/hyperparam-opt.ipynb (hunk 1 above)
Comment (stsievert) [reply machine-learning/hyperparam-opt.ipynb]: Whoops, thanks.

File: machine-learning/hyperparam-opt.ipynb (hunk 2 above)
Comment (stsievert) [reply machine-learning/hyperparam-opt.ipynb]: Resolved.

**Comment by TomAugspurger** [comment]
Sounds good. Thanks for putting this together!

**Pull request merged**